src/middlewares/createRouterMiddleware.ts

```diff
@@ -57,10 +58,6 @@ export const createRouterMiddleware = <
     // casting to UiState here to keep createURL unaware of custom UiState
     // (as long as it's an object, it's ok)
     instantSearchInstance._createURL = topLevelCreateURL as CreateURL<UiState>;
-    instantSearchInstance._initialUiState = {
-      ...instantSearchInstance._initialUiState,
-      ...stateMapping.routeToState(router.read()),
-    };
 
```

Comment:
in that case the initial ui state can be captured here instead of making it an argument @eunjae-lee

Reply:
yeah good point!